### DISCUSSION TITLE
What's the best practice on collection relations?

### DISCUSSION THREAD

**Discussion by Plorenzo** [pocketbase/pocketbase#4680] [Q&A] [answered]
So I have some user data that want to make public, as currently there is no way to specify permissions to specific fields in a collection the recommended way is to create another collection. `Users` collection then will be protected by `@request.auth.id = user.id ` and the public collection `UsersPublic` won't have any restrictions. 

Now, my question is what is the recommended way to set this up? Create UsersPublic with `field1`, `field2` and then add a `UsersPublic` field in `Users` or the other way around create UsersPublic with `Users`, `field1` and `field2` ? 

My first approach was to do it the 2nd way, having the public `UsersPublic` have the relation but then it becomes inconvenient expand the relationship, I can't do expand: UsersPublic, I need to make another call to `UsersPubilc` or use indirect expands which I find more difficult to understand. 

I know pocketbase is not using foreign keys under the hood, so I would like to understand if there is a preferred way when arranging this kind or relations (beside my specific example: main collection having pointers to child collections or child collections have pointer to parent collection) or if there is any performance implications that I need to be aware of.

**Comment by ganigeorgiev** [comment, marked as answer]
> I can't do expand: UsersPublic, I need to make another call to UsersPubilc or use indirect expands which I find more difficult to understand

I don't understand from where the confusion comes from. I want to make it more simple and clear but I don't know how.

To reference indirect/back-relations the syntax is `relCollection_via_relField` where `relField` is the field that will be used to join/expand `relCollection` into your main one. For your example if the name of the relation field is "user" this would be:

```js
await pb.collection("Users").getFullList({
    // see https://pocketbase.io/docs/working-with-relations/#back-relations
    expand: "UsersPublic_via_user"
})
```

Regarding on which side to hold the relation field - the same would apply if you were using a more traditional database setup (_whether the foreign key constraint is checked on app or db level has nothing to do with that_). It is difficult to generalize and it will depend on your specific use case and desired access patterns. I don't expect performance difference between the two options but again it depend on your actual use case. Storing the relation field inside `Users` could allow a single `UsersPublic` to be used by multiple users but it would also require extra access checks to prevent a user to assign to their account some other `UsersPublic` record. Based only on the description and without having context about the full picture of your use case, I would maybe go the other way - having the relation field in the `UsersPublic` collection with a Unique index on the relation field to prevent a single user to have multiple "UsersPublic" records.
Additionally note that I would usually do the opposite of your example, aka. I would keep the "public" safe fields inside the `users` collection and move the private/hidden fields to a another collection (_if using a simple `@request.data.*` filter is not sufficient or too bothersome in case there too many private/hidden fields_).

**Reply by Plorenzo** [reply]
> I don't understand from where the confusion comes from. I want to make it more simple and clear but I don't know how.

Now reading your explanation is crystal clear, but whenever I'm developing and creating new collections I always struggle to get the indirect relation right in my head, not sure why also. 

Anyway, thanks for the advice! I will keep the relation field on the _child_ collections for the moment and see how it goes
